### ISSUE TITLE
undefined symbol: EVP_MD_CTX_reset

### ISSUE THREAD

**Issue by abhi98041** [jpmens/mosquitto-auth-plug#193]
I am getting an error whenever I am trying to run this file.
Run Command:  /usr/local/sbin/mosquitto -c /etc/mosquitto/mosquitto.conf

Error
```
1484564338: mosquitto version 1.4.10 (build date 2017-01-14 03:58:18-0800) starting
1484564338: Config loaded from /etc/mosquitto/mosquitto.conf.
1484564338: Error: Unable to load auth plugin "/etc/mosquitto/auth-plug.so".
1484564338: Load error: /etc/mosquitto/auth-plug.so: undefined symbol: EVP_MD_CTX_reset
```

**Comment by douglaszuqueto** [comment]
I have the same problem.

**Comment by medaboub** [comment]
I have exactly the same problem
please some one help.

**Comment by JoshuaCurry** [comment]
Can confirm same issue - OpenSSL V1.1.0d compiled/installed from git running on RHEL 7.0

**Comment by douglaszuqueto** [comment]
I solved my problem through this pr: https://github.com/jpmens/mosquitto-auth-plug/pull/184

**Comment by eddiehd** [comment]
I am currently having the same issue 
2017-01-11 Raspbian-jessie-lite

/usr/local/sbin/mosquitto -c /etc/mosquitto/mosquitto.conf -v
1487044649: mosquitto version 1.4.10 (build date 2017-02-13 02:27:00+0000) starting
1487044649: Config loaded from /etc/mosquitto/mosquitto.conf.
1487044649: Error: Unable to load auth plugin "/etc/mosquitto/auth-plug.so".
1487044649: Load error: /etc/mosquitto/auth-plug.so: undefined symbol: EVP_MD_CTX_reset

OpenSSL 1.1.0  25 Aug 2016

 ldd auth-plug.so
        linux-vdso.so.1 (0x7ecce000)
        /usr/lib/arm-linux-gnueabihf/libarmmem.so (0x76efc000)
        libmysqlclient.so.18 => /usr/lib/arm-linux-gnueabihf/libmysqlclient.so.18 (0x76be6000)
        libpthread.so.0 => /lib/arm-linux-gnueabihf/libpthread.so.0 (0x76bbe000)
        libz.so.1 => /lib/arm-linux-gnueabihf/libz.so.1 (0x76b97000)
        libm.so.6 => /lib/arm-linux-gnueabihf/libm.so.6 (0x76b1c000)
        libdl.so.2 => /lib/arm-linux-gnueabihf/libdl.so.2 (0x76b09000)
        libcrypto.so.1.0.0 => /usr/lib/arm-linux-gnueabihf/libcrypto.so.1.0.0 (0x7699a000)
        libc.so.6 => /lib/arm-linux-gnueabihf/libc.so.6 (0x76859000)
        libstdc++.so.6 => /usr/lib/arm-linux-gnueabihf/libstdc++.so.6 (0x7677d000)
        libgcc_s.so.1 => /lib/arm-linux-gnueabihf/libgcc_s.so.1 (0x76750000)
        /lib/ld-linux-armhf.so.3 (0x54b9c000)

WITH_SRV=no

I also tried following pr:184 [https://github.com/jpmens/mosquitto-auth-plug/pull/184](url) but no luck.

What am I missing here?

**Comment by jpmens** [comment]
Fixed via #184
